src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsEmulatorConfigWriterHook.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using System.Text.Json;
+
+namespace Aspire.Hosting.Azure;
+
+internal sealed class AzureEventHubsEmulatorConfigWriterHook : IDistributedApplicationLifecycleHook
+{
+    public Task AfterEndpointsAllocatedAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken)
+    {
+        if (appModel.Resources.OfType<AzureEventHubsResource>().SingleOrDefault() is not { } emulatorResource || !emulatorResource.IsEmulator)
```

Comment:
> SingleOrDefault

Why do we think there will be only one?

Reply:
Got this code from Redis and PhpMyAdmin. I suggest to replace by `FirstOrDefault` in all places, seems good?

Reply:
Done

Reply:
Why not a loop? Can't I use multiple EventHubs in my app?

Reply:
Yes, it's a loop now, refresh.